modules/arm_plugin/src/arm_converter/arm_converter_quantize.cpp

```diff
@@ -9,10 +9,128 @@
+        _quant->configure(input, &_outputqi);
+    }
+    static arm_compute::Status validate(const arm_compute::ITensorInfo *input, const arm_compute::ITensorInfo *output,
+                                        const arm_compute::QuantizationInfo &qi) {
+        ARM_COMPUTE_RETURN_ERROR_ON_NULLPTR(output);
+        //At the moment quantization info isn't checked actually, but just in case
+        return arm_compute::NEQuantizationLayer::validate(input, &arm_compute::TensorInfo(*output).set_quantization_info(qi));
+    }
+    void run() override {
+        ARM_COMPUTE_ERROR_ON_MSG(!_quant.get(), "Kernel didn't configured");
+        if (_outputqi.info()->padding() != _output->info()->padding()) _outputqi.info()->extend_padding(_output->info()->padding());
+        _outputqi.allocator()->import_memory(_output->buffer());
+        _quant->run();
+        _outputqi.allocator()->free();
+    }
+
+protected:
+    const arm_compute::ITensor *_output;
+    arm_compute::Tensor _outputqi;
+    std::unique_ptr<arm_compute::NEQuantizationLayer> _quant;
+};
+template <typename T, typename Q>
+void quantize(const T* input_data,
+              Q* output_data,
+              const ngraph::Shape& input_data_shape,
+              float scale,
+              std::int32_t offset) {
+    for (size_t i = 0; i < ov::shape_size(input_data_shape); ++i) {
+        float output = input_data[i] / scale + offset;
+        output -= std::remainder(output, 1.0f);
+        if (output < std::numeric_limits<Q>::lowest()) {
+            output_data[i] = std::numeric_limits<Q>::lowest();
+        } else if (output > std::numeric_limits<Q>::max()) {
+            output_data[i] = std::numeric_limits<Q>::max();
+        } else {
+            output_data[i] = static_cast<Q>(output);
+        }
+    }
+}
 template<> Converter::Conversion::Ptr Converter::Convert(const opset::ArmQuantize& node) {
-    return MakeConversion<arm_compute::NEQuantizationLayer>(node.input(0), node.output(0));
+    auto qInfoIt = node.get_rt_info().find("QuantizationInfo");
+    if (qInfoIt == node.get_rt_info().end()) {
+        IE_THROW() << "No quantization info available for ArmQuantize";
+    }
+    auto qInfo = safe_cast<ngraph::VariantWrapper<arm_compute::QuantizationInfo>>(qInfoIt->second)->get();
+    if (node.get_input_element_type(0).is_real()) {
+        auto make = [&] (auto refFunction) {
+            return this->MakeConversion(refFunction,
+                                        node.input(0),
+                                        node.output(0),
+                                        node.get_input_shape(0),
+                                        qInfo.scale()[0],
+                                        qInfo.offset()[0]);
+        };
+        return CallSwitch(
+            AP_WRAP(make, quantize),
+            node.get_input_element_type(0), floatTypes,
+            node.get_output_element_type(0), intTypes);
```

Comment:
Do we need quantization into all integer types?

Reply:
fixed